### DISCUSSION TITLE
🗂【面试题】技术面试题汇总

### DISCUSSION THREAD

**Discussion by imageslr** [imageslr/blog#85] [Announcements]
https://imageslr.github.io/2020/07/08/tech-interview.html 

 这是我用来准备后端开发校招面试的笔记汇总。可以用作复习阶段的知识点梳理，也可以用作面试前的快速回顾。 持续更新中…

**Comment by zhangxu-ai** [comment]
谢谢楼主的笔记！很好！

**Comment by imageslr** [comment]
> 谢谢楼主的笔记！很好！

最近几个月忘记更新了，争取今年秋招前更新完！

**Comment by huanting0016** [comment]
赞，今年冲大厂就靠楼主了

**Comment by imageslr** [comment]
> 赞，今年冲大厂就靠楼主了

这两天我先把去年秋招遇到的题目都更新上去，这样大家可以先在网上搜答案。如果那个需要优先展开讲，可以艾特我。

**Comment by huanting0016** [comment]
> @imageslr
> > 赞，今年冲大厂就靠楼主了
> 
> 这两天我先把去年秋招遇到的题目都更新上去，这样大家可以先在网上搜答案。如果那个需要优先展开讲，可以艾特我。

可以可以，辛苦楼主了，基础过于薄弱，需要好好学习一下

**Comment by midasism** [comment]
很详细，先收藏了

**Comment by imageslr** [comment]
> @LydiaCai1203
> 第二次机会置换算法写的是页面访问两次就被置换，这么描述好像不太对。

谢谢指正，这部分确实照抄了本科笔记，没有深入探究。我之后会根据你提供的资料，再改正一下！

**Comment by imageslr** [comment]
有更好的解答，欢迎[发个 PR](https://github.com/imageslr/blog/edit/master/_posts/2020-07-08-tech-interview.md) 👍

**Comment by jingxuantju** [comment]
想问一下楼主 同一进程的线程切换会进入内核态么？ 谢谢了

**Comment by imageslr** [comment]
> 想问一下楼主 同一进程的线程切换会进入内核态么？ 谢谢了

会。只有内核才有权力进行进程 / 线程调度。

**Comment by jingxuantju** [comment]
> @imageslr
> > 想问一下楼主 同一进程的线程切换会进入内核态么？ 谢谢了
> 
> 会。只有内核才有权力进行进程 / 线程调度。
> 

谢谢楼主！ 学习了 明年找楼主内推 ☺

**Comment by imageslr** [comment]
Ads Infra 2023 届后端实习开始内推啦，可转正，欢迎加入👏🏻 https://www.nowcoder.com/discuss/834943

**Comment by Vector6662** [comment]
Linux常用命令板块中的一个题目：*如何统计一个文件中出现次数最多的前 10 个单词？* 我觉得答案有些问题。正如您对sort的解释一样，是以行为单位，所以用了sort只能对每行的**开头**单词就行排序。而且uniq的去重单位也是行。

**Comment by imageslr** [comment]
@Vector6662 谢谢指正。这里假设 words.txt 是每行一个单词。如果 words.txt 是多个单词在同一行、用空格分隔，我们可以使用 `sed` 命令来将其改成每行一个单词。比如：

words.txt：
```
apple banana orange
```
执行：
```
cat words.txt | sed 's/ /\n/g'
```

输出：

```
apple
banana
orange
```

**Comment by gitzhangjz** [comment]
海量数据问题有个地方是不是写错了：“如果题目给定数据大小范围是 0～10000，那么只需要申请一个 10000 字节的空间。”应该是需要申请10000bit的空间。

**Reply by imageslr** [reply]
是的，谢谢指正。
